lib/adagrams.rb

```diff
@@ -0,0 +1,115 @@
+require 'awesome_print'
+require 'pry'
+
+def draw_letters
+    letters = [
+        "A", "A", "A", "A", "A", "A", "A", "A", "A", 
+        "B", "B",
+        "C", "C",
+        "D", "D", "D", "D",
+        "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E",
+        "F", "F",
+        "G", "G", "G",
+        "H", "H",
+        "I", "I", "I", "I", "I", "I","I", "I", "I",
+        "J",
+        "K",
+        "L", "L", "L", "L",
+        "M", "M",
+        "N", "N", "N", "N", "N", "N",
+        "O", "O", "O", "O", "O", "O", "O", "O",
+        "P", "P",
+        "Q",
+        "R", "R", "R", "R", "R", "R",
+        "S", "S", "S", "S",
+        "T", "T", "T", "T", "T", "T",
+        "U", "U", "U", "U",
+        "V", "V",
+        "W", "W",
+        "X",
+        "Y", "Y",
+        "Z"
+    ]
+    drawn_letters = letters.sample(10)
+    return drawn_letters
+end
+
+
+draw_letters
+
+def uses_available_letters?(input, letters_in_hand)
+    input.each_char do |letter|
+        index = letters_in_hand.index(letter)
+        if index.nil?
+            return false
+        else
+            letters_in_hand.delete_at(index)
+        end
+    end
+    return true
+end
+
+def score_word (word)
+  score = 0
+  word.upcase.each_char do |letter|
+  
+  case letter
+  when "A", 'E', 'I', 'O', 'U', 'L', 'R', 'N', 'S', 'T' 
+    score += 1
+  when 'D', 'G'
```

Comment:
While this `case` statement works, it means that the information about which letter has which score is locked into this piece of code, and can't easily be used elsewhere. For example, if you wanted to display the value of each letter in a hand, you would need to repeat this work.

An alternative approach would be to store the letter scores in a hash, something like this:

```ruby
LETTER_SCORES = {
  "A" => 1
  "B" => 3,
  "C" => 3,
  "D" => 2,
  # ...
}
```

Then to get the score for a letter, you can say `LETTER_SCORES[letter]`.